src/Element/EntityAutocompleteMultiple.php

```diff
@@ -0,0 +1,277 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_corporate_site_info\Element;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Component\Utility\SortArray;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\FormElement;
+
+/**
+ * Provides an multiple entity autocomplete form element.
+ *
+ * The autocomplete form element allows users to select one or multiple
+ * entities, which can come from all or specific bundles of an entity type.
+ * Some features and behavior of this form element inherited from
+ * the Autocomplete widget.
+ *
+ * @see \Drupal\Core\Field\Plugin\Field\FieldWidget\EntityReferenceAutocompleteWidget
+ *
+ * Properties:
+ * - #add_more_title: (optional) Title of the "add more" button.
+ *
+ * Usage example:
+ * @code
+ * $form['my_element'] = [
+ *  '#type' => 'oe_corporate_site_info_entity_autocomplete_multiple',
+ *  '#target_type' => 'node',
+ *  '#default_value' => [
+ *    $entity,
+ *  ],
+ *  '#selection_handler' => 'default',
+ *  '#selection_settings' => [
+ *    'target_bundles' => ['article', 'page'],
+ *   ],
+ * ];
+ * @endcode
+ *
+ * @see \Drupal\Core\Entity\Plugin\EntityReferenceSelection\DefaultSelection
+ * @see \Drupal\Core\Entity\Element\EntityAutocomplete
```

Comment:
Actually a lot of it comes from WidgetBase, let's put a reference here.

Reply:
done